src/Core/Geographic/GeoidGrid.js

```diff
@@ -0,0 +1,131 @@
+ * //         40.0  |   1  2  3
+ * //               |------------->
+ * //                   1  2  3     Longitudes
+ *
+ * // Create a GeoidGrid allowing to access the gridded data.
+ * const geoidGrid = new GeoidGrid(
+ *     new Extent('EPSG:4326', 1, 3, 40, 41),
+ *     new THREE.Vector2(1, 0.5),
+ *     (verticalIndex, horizontalIndex) => data[verticalIndex][horizontalIndex],
+ * );
+ *
+ * // Access a value of geoid height at some geographic coordinates.
+ * // The value is interpolated from the gridded data.
+ * const value = geoidGrid.getGeoidAtCoordinates(
+ *     new Coordinates('EPSG:4326', 1.5, 40.25)
+ * );
+ * // This should return 2.0, which is the result from the bi-linear
+ * // interpolation at the center of the `[[1, 2], [2, 3]]` subsection
+ * // of the grid data.
+ */
+class GeoidGrid {
+    /**
+     * @param   {Extent}        extent              The geographic extent of the geoid height grid data.
+     * @param   {THREE.Vector2} step                The distance between two consecutive points of the geoid height
+                                                    * grid. The `x` value stands for the distance along the West-East
+                                                    * direction, and the `y` value stands for the distance along the
+                                                    * South-North direction.
+     * @param   {function}      dataReadingMethod   A method that allows reading a value in the geoid height grid from
+                                                    * its vertical and horizontal indexes. The lower an index, the lower
+                                                    * the coordinate on the corresponding axis - 0 being the index of
+                                                    * the minimal coordinate of the gridded data on a given axis.
+     */
+    constructor(
+        extent,
+        step,
+        dataReadingMethod,
+    ) {
+        CRS.isGeographic(extent.crs);
+
+        this._extent = extent;
+        this._step = new THREE.Vector2(step.x, step.y || step.x);
+
+        this._dimensions = this._extent.planarDimensions();
+        this._dataSize = new THREE.Vector2().addVectors(this._step, this._dimensions).divide(this._step).round();
+
+        this._readDataAtIndexes = dataReadingMethod;
+    }
+
+    /**
+     * Get the value of the geoid height at given geographic `{@link Coordinates}`. The geoid height value is
+     * bi-linearly interpolated from the gridded data accessed by the `GeoidGrid` instance.
+     *
+     * @param   {Coordinates}   coordinates     Geographic coordinates to get the geoid height value at.
+     *
+     * @returns {number}    The geoid height value at the given `{@link Coordinates}`, bi-interpolated from the gridded
+                            * data accessed by the `GeoidGrid` instance.
+     */
+    getGeoidAtCoordinates(coordinates) {
+        coordinates.as(this._extent.crs, coord);
+
+        const indexes = new THREE.Vector2(
```

Comment:
move out class the `Vector2` instancing

Reply:
I now instantiate `Vector2` outside of the method.